.github/workflows/testing.yml

```diff
@@ -0,0 +1,159 @@
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+  workflow_dispatch:
+    type: choice
+
+env:
+  # Indicated the testing script, that it is performed on GitLab. In this
+  # case all test have to pass.
+  TESTING_GITLAB: 1
+  # Some test require baci-release to check if the created input file
+  # works with baci.
+  BACI_BUILD: "/hdd/gitlab-runner/lib/baci-master/release"
+  BACI_RELEASE: "$BACI_BUILD/baci-release"
+  # Python executable and virtual environment name
+  PYTHON_EXE: "python3"
+  PYTHON_VENV: "python-testing-environment"
+  # Meshpy can interact with cubitpy, to perform the corresponding tests,
+  # the following paths have to be set
+  CUBITPY_REPO: "/hdd/gitlab-runner/lib/cubitpy-master"
+  BACI_PRE_EXODUS: "$BACI_BUILD/pre_exodus"
+  CUBIT: "/imcs/public/compsim/opt/cubit-15.2"
+  # Meshpy allows to perform geometric search functions with ArborX,
+  # for this functionality we need Kokkos. On IMCS Ares this is provided
+  # with the spack installation under the following path.
+  SPACK_ACTIVATION_SCRIPT: "/hdd/gitlab-runner/lib/spack/share/spack/setup-env.sh"
+  # For building the website.
+  JEKYLL_ENV: production
+
+jobs:
+  meshpy-testing:
+    name: meshpy-testing
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Test meshpy
+        run: |
+          whoami
+          pwd
+          MESHPY_PATH=$(pwd)
+          # Create the virtual environment
+          ${PYTHON_EXE} -m venv ${PYTHON_VENV}
+          source ${PYTHON_VENV}/bin/activate
+          # We have to update pip, otherwise there is a bug in the compilation of the cython code
+          pip install --upgrade pip
+          # Install meshpy
+          pip install .[CI-CD]
+          # Install cubitpy
+          cd $CUBITPY_REPO
+          git checkout master
+          git pull
+          pip install .
```

Comment:
The repository of cubitpy seems to cause issues. You can try to simply pull cubitpy from the github repostory here (I would assume the runner has access rights).

This should look something like this:
```
cd $MESHPY_PATH
git clone git@github.com:imcs-compsim/cubitpy.git
pip install cubitpy/
```